src/common/LogClient.h

```diff
@@ -209,7 +209,12 @@ class LogClient
     return create_channel(CLOG_CHANNEL_DEFAULT);
   }
 
-  LogChannelRef create_channel(const std::string& name) {
+  LogChannelRef create_channel(std::string name) {
```

Comment:
is `create_channel()` a public API? why shall we use DEFAULT if name does not match any of the known channel?

Reply:
As I checked, it would create a new log channel with an unknown name and add it to the map of `channels` otherwise.

Reply:
i am not sure if the questions are answered. i mean, it'd be bug if some of us uses a name for creating a new channel. and this is expected. your change covers this bug.

Reply:
There is no bug at present. I've noticed the `create_channel()` code while trying to use it and also the [channels](https://github.com/ceph/ceph/blob/master/src/common/LogEntry.h#L27) defined. It seemed like we don't want people to pass any string and create a channel named say `xyz` (For instance, via ActivePyModules::cluster_log()), as these are not standard log channel names.

If this patch doesn't make sense, I'm okay closing it. Let me know.

Reply:
i don't think this change makes sense. as i don't think `create_channel()` is a public API. if we find that there are a bunch of log channels get created and those channel names are just gibberish or random strings. i think it's a bug. so far the only place where a user specified channel name is allowed is `ActivePyModules::cluster_log()`. but i think that's by-design.

what do you think?